### PR TITLE
[Feature] Support Changing the bucketing mode of the table from Hash Distribution to Random Distribution

### LINKED ISSUE #8258: [Feature]  Support Changing the bucketing mode of the table from Hash Distribution to Random Distribution
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Description

_No response_

### Use case

 ALTER TABLE example_db.my_table set ("distribution_type" = "random");

### Related issues

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by caiconghui**
# Proposed changes

Issue Number: close #8258 

## Problem Summary:

Describe the overview of changes.

## Checklist(Required)

1. Does it affect the original behavior: (Yes/No/I Don't know)
2. Has unit tests been added: (Yes/No/No Need)
3. Has document been added or modified: (Yes)
4. Does it need to update dependencies: (No)
5. Are there any changes that cannot be rolled back: (No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/catalog/HashDistributionInfo.java [hunk 1]
```diff
@@ -146,4 +146,9 @@ public String toString() {
 
         return builder.toString();
     }
+
+    public RandomDistributionInfo toRandomDistributionInfo() {
+        RandomDistributionInfo randomDistributionInfo = new RandomDistributionInfo(bucketNum);
```
Comment (levy5307) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/HashDistributionInfo.java:151]:
```
       RandomDistributionInfo randomDistributionInfo = new RandomDistributionInfo(bucketNum);
        return randomDistributionInfo;
```
==>
```
return new RandomDistributionInfo(bucketNum);
```

File: fe/fe-core/src/main/java/org/apache/doris/catalog/Catalog.java [hunk 2]
```diff
@@ -7043,12 +7036,12 @@ public void replayBackendTabletsInfo(BackendTabletsInfo backendTabletsInfo) {
     public void convertDistributionType(Database db, OlapTable tbl) throws DdlException {
         tbl.writeLockOrDdlException();
         try {
-            if (!tbl.convertRandomDistributionToHashDistribution()) {
-                throw new DdlException("Table " + tbl.getName() + " is not random distributed");
+            if (!tbl.convertHashDistributionToRandomDistribution()) {
```
Comment (levy5307) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/Catalog.java:7039]: Could you add some unit tests for your changes?

File: fe/fe-core/src/main/java/org/apache/doris/catalog/HashDistributionInfo.java (hunk 1 above)
Comment (caiconghui) [reply fe/fe-core/src/main/java/org/apache/doris/catalog/HashDistributionInfo.java]: ok

File: fe/fe-core/src/main/java/org/apache/doris/catalog/Catalog.java (hunk 2 above)
Comment (caiconghui) [reply fe/fe-core/src/main/java/org/apache/doris/catalog/Catalog.java:7038]: done

**Review by weizuo93 — DISMISSED** [review]
LGTM

File: fe/fe-core/src/main/java/org/apache/doris/analysis/ModifyTablePropertiesClause.java [hunk 3]
```diff
@@ -55,7 +55,10 @@ public void analyze(Analyzer analyzer) throws AnalysisException {
                 throw new AnalysisException("Can only change storage type to COLUMN");
             }
         } else if (properties.containsKey(PropertyAnalyzer.PROPERTIES_DISTRIBUTION_TYPE)) {
-            throw new AnalysisException("Cannot change distribution type for olap table now");
+            if (!properties.get(PropertyAnalyzer.PROPERTIES_DISTRIBUTION_TYPE).equalsIgnoreCase("random")) {
+                throw new AnalysisException("Can only change distribution type from RANDOM to HASH");
```
[the remark this answers is not in the document]
Comment (caiconghui) [reply fe/fe-core/src/main/java/org/apache/doris/analysis/ModifyTablePropertiesClause.java:59]: my fault, I will fix the error message

**Comment by yiguolei** [comment]
LGTM

**Review by weizuo93 — APPROVED** [review]

**Pull request merged**